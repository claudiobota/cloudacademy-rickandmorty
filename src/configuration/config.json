{
	"api": {
		"url": "https://rickandmortyapi.com/api",
		"paths": {
			"characters": "/character",
			"locations": "/location",
			"episodes": "/episode"
		},
		"page_size": 20
	}
}
