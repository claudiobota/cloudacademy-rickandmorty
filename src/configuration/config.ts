import configData from './config.json';
import configDevelopmentData from './config.development.json';
import configTestData from './config.test.json';
import configStagingData from './config.staging.json';
import configProductionData from './config.production.json';
import {IConfig} from '../types/interfaces';
import {Environment} from '../types/enums';

let config: IConfig = {
  ...configData
};

const nodeEnv: string = process.env.NODE_ENV || 'development';

switch (nodeEnv) {
  case Environment.DEVELOPMENT:
    config = {
      ...config,
      ...configDevelopmentData
    };
    break;

  case Environment.TEST:
    config = {
      ...config,
      ...configTestData
    };
    break;

  case Environment.STAGING:
    config = {
      ...config,
      ...configStagingData
    };
    break;

  case Environment.PRODUCTION:
    config = {
      ...config,
      ...configProductionData
    };
    break;

  default:
}

export default config;
