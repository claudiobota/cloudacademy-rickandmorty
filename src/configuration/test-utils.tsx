import React from 'react';
import {render as rtlRender} from '@testing-library/react';
import {Provider} from 'react-redux';
import {configureAppStore} from '../redux/store';

function render(
  ui: React.ReactElement,
  {
    initialState,
    store = configureAppStore(initialState),
    ...renderOptions
  }: any = {}
) {
  function Wrapper({ children }: { children: React.ReactElement }) {
    return <Provider store={store}>{children}</Provider>;
  }

  return rtlRender(ui, { wrapper: Wrapper, ...renderOptions });
}

export * from '@testing-library/react';
export { render }
