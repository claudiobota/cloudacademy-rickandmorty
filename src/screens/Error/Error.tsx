import React from 'react';

export default function Error() {
  return (
    <div className="container p-text-center p-text-uppercase">
      <h1 style={{ color: '#FFFFFF', fontSize: '50px' }} className="p-mb-1">404</h1>
      <h2 style={{ color: '#FFFFFF', opacity: 0.5 }} className="p-mt-0">Page not found</h2>
    </div>
  )
}
