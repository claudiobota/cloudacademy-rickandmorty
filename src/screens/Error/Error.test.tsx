import React from 'react';
import Error from './Error';
import {render} from '../../configuration/test-utils';

describe('Error', () => {
  test('screen renders and matches snapshot', () => {
    const {asFragment} = render(<Error />);
    expect(asFragment()).toMatchSnapshot();
  });
});
