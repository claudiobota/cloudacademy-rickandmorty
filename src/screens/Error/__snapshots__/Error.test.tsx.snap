// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Error screen renders and matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="container p-text-center p-text-uppercase"
  >
    <h1
      class="p-mb-1"
      style="color: rgb(255, 255, 255); font-size: 50px;"
    >
      404
    </h1>
    <h2
      class="p-mt-0"
      style="color: rgb(255, 255, 255); opacity: 0.5;"
    >
      Page not found
    </h2>
  </div>
</DocumentFragment>
`;
